File: Workshops/24-06-26 - Cards/eCommerce/UI/ecommerce_mobile/lib/models/payment_card_ib180079.dart
import 'package:json_annotation/json_annotation.dart';

part 'payment_card_ib180079.g.dart';

@JsonSerializable()
class PaymentCardIB180079 {
  final int id;
  final int userId;
  final String cardNumber;
  final String cvc;
  final DateTime exiprationDate;
  final double initialBalance;

  PaymentCardIB180079({
    required this.id,
    required this.userId,
    required this.cardNumber,
    required this.cvc,
    required this.exiprationDate,
    required this.initialBalance,
  });

  factory PaymentCardIB180079.fromJson(Map<String, dynamic> json) =>
      _$PaymentCardIB180079FromJson(json);

  Map<String, dynamic> toJson() => _$PaymentCardIB180079ToJson(this);
}

File: Workshops/1. 24-06-26 - Cards/eCommerce/eCommerce.Services/Database/PaymentCardIB180079.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eCommerce.Services.Database
{
    public class PaymentCardIB180079
    {
        [Key]
        public int Id { get; set; }
        
        // Foreign key for User
        public int UserId { get; set; }
        
        // Navigation property for Product
        [ForeignKey("UserId")]
        public User User { get; set; } = null!;


        [Required]
        [MaxLength(12)]
        public string CardNumber { get; set; } = string.Empty;

        [Required]
        [MaxLength(3)]
        public string CVC { get; set; } = string.Empty;

        public DateTime ExiprationDate { get; set; } 

        public decimal InitialBalance { get; set; }
        
        
        
    }
} 